fitbenchmarking/fitbenchmarking
# Enable CT for release branch

#### Description of Work
- Added whitelist to .travis.yml with master and a regex for all release branches (matches anything starting with "release-"
- Added notifications to fitbenchmarking slack channel (build-failures)

Two notes:
- I realised that adding a badge might not be practical, as it would need to continually be updated to point at the latest release to be of any use.

If you have any thoughts on these then please let me know.

**NB: To complete this PR, Travis has had "Build on push" enabled in the settings.  I think that because other branches will not have a whitelist in .travis.yml, they will build on every push, which is not desired.  Therefore I recommend that, after this PR is merged, every current branch should have master merged as a priority, to stop a lot of unnecessary builds.**

Fixes #574 


#### Testing Instructions

1. The only practical test will be when this is merged to master (a push build on master should appear on Travis) and then again when it is cherry-picked to the release (again, push build on Travis).
2. Other than this, if you wanted to be sure that the slack notification was working, after merging you could either:
- add a commit to master or release which breaks the build and see if slack gets a notification,
- or temporarily change the slack notification conditions to `success: always`

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
